src/HttpClientInterception/HttpRequestInterceptionBuilder.cs

```diff
@@ -79,7 +80,7 @@ public HttpRequestInterceptionBuilder()
         /// </remarks>
         public HttpRequestInterceptionBuilder For(Predicate<HttpRequestMessage> predicate)
         {
-            _requestMatcher = predicate == null ? null : DelegateHelpers.ConvertToBooleanTask(predicate);
+            _requestMatcher = predicate == null ? null : new Func<HttpRequestMessage, Task<bool>>((message) => Task.FromResult(predicate(message)));
```

Comment:
Should this be using one of those new overloads?

Reply:
I had trouble with this one, the additional overloads mean that here the call becomes ambiguous, it's the only place I have to give it a type.

Reply:
Hmm. I wonder if there's any naming or something that can be changed to make it non-ambiguous. The `DelegateHelpers` stuff was to try and get away from the need to do this stuff inline.

Reply:
I was considering inline all of these to be honest, there are enough variations of trivial delegate conversions going on it seems it would be more straightforward to just do what needs doing inline.

The alternative is separate out the methods in `DelegateHelpers` to be more fine grained.

Let me know which way you want to go.

Reply:
I think _part_ of the reason I put this here was to reduce the amount of null checks before creating the delegate that would have been needed at all the call sites.